packages/babel-helper-module-transforms/src/rewrite-live-references.js

```diff
@@ -259,13 +263,14 @@ const rewriteReferencesVisitor = {
           );
           requeueInParent(path);
         }
-      } else if (left.isMemberExpression()) {
-        // No change needed
       } else {
         const ids = left.getOuterBindingIdentifiers();
-        const id = Object.keys(ids)
-          .filter(localName => imported.has(localName))
-          .pop();
+        const programScopeIds = Object.keys(ids).filter(
+          localName =>
+            scope.getBinding(localName) === path.scope.getBinding(localName),
```

Comment:
nit: consider to guard with `scope.hasBinding` to avoid unnecessary `path.scope.getBinding`.

Reply:
I tried to do it and found a bug which, if fixed, would break `@babel/plugin-transform-runtime` <= 7.6 :cry: 

This is what currently happens:
1) The imports are analyzed
2) The imports are removed -> The bindings are removed
3) References to imported bindings are rewritten (this file)
    i) When we are comparing the bindings, the global one is `undefined`
    ii) If the local binding is `undefined`, it hasn't been redefined.

I tried to swap 2 and 3, but I still can optimize away the check for the local binding because `@babel/transform-runtime` doesn't register the import it injects in the scope tracker.